### ISSUE TITLE
Open database failed

### ISSUE THREAD

**Issue by jfreak53** [brody2consult/cordova-sqlite-storage#499]
I've installed the latest version in my PhoneGap app using this command `phonegap plugin add https://github.com/litehelpers/Cordova-sqlite-storage`. I then did a `phonegap platform rm android` and then `phonegap platform add android` to make sure the plugin was there.

I started up my app in `phonegap serve` and tried to open/create a new database and it's telling me that the database can't be opened. This is the error:

```
[phonegap] [console.log] OPEN database: syncdata.db
[phonegap] [console.log] { openargs: 
   { name: 'syncdata.db',
     location: 'default',
     dblocation: 'nosync' },
  dbname: 'syncdata.db' }
[phonegap] [console.log] OPEN database: syncdata.db failed, aborting any pending transactions
[phonegap] [console.error] { code: 0 }
```

This is the open code I'm using:

```
db = window.sqlitePlugin.openDatabase({name: 'syncdata.db', location: 'default' }, populateDB, errorCB);
console.log(db);
```

I've tried it also with `androidDatabaseImplementation: 2` and still the same error. I'm using PhoneGap version 6.2.6.

**Comment by brody2consult** [comment]
It sounds like you are trying to use the browser platform or PhoneGap Developer App, neither of which are supported for this plugin. I will document this better when I get a chance.

**Comment by jfreak53** [comment]
Browser or Developer App? I'm lost. This is a Phonegap CLI project that I serve for testing from the cli using `phonegap serve` then build using `phonegap build` and launch into an apk for android only. I'm not using a browser nor the developer app that I know of :(

**Comment by brody2consult** [comment]
OK. PhoneGap CLI is not officially supported by this project either. But if you can build and install it, it should work.

Is this a new app or is this an upgrade?

Did you try the echoTest and selfTest functions as documented?

**Comment by jfreak53** [comment]
Upgrade, was using the built in webSQL standard for PG but opted for this plugin instead since webSQL is now deprecated.

Based on what you just said though I don't think we will be using it :( I can't believe that to test this app using this plugin now I have to Build it each and every time. Wow, cumbersome to say the least haha. If you can't use this plugin in a RAD testing environment then it's very cumbersome to test and debug. It requires a Build to APK each time, then re-install on a device or emulated environment. That is slow and cumbersome.

Thanks but that just answered my question, gonna have to look for another solution :(..Thanks for responding so fast :)

**Comment by brody2consult** [comment]
I understand. This plugin is missing support for the browser platform which could make it much easier. I did list some other alternatives in the README so you are welcome to look through it.

Just a quick comment that you did not say whether or not your app waits for the deviceready event before attempting to open the database. This is necessary for all Cordova plugins. I do find this a bit cumbersome, especially for Ionic platform users.

**Comment by Rombersoft** [comment]
I've taken example from https://www.npmjs.com/package/cordova-plugin-sqlite
Note that I use database working after 'deviceready' event but I get same problem. I don't have database but I want to create it. I don't use phoneGap and use Cordova.
I assume that when project is underdeveloped or does not have understandable documentation with described methods and examples it have to be closed and not clog up our brains. Are you agree?

function onDeviceReady() {
    if( navigator.splashscreen && navigator.splashscreen.hide ) {   // Cordova API detected
        navigator.splashscreen.hide() ;
    }
    document.getElementById("save").addEventListener("click", OnSave);
    document.getElementById("open").addEventListener("click", OnOpen);
}

function OnSave()
{
    var db = window.sqlitePlugin.openDatabase({name: "my.db"});
 
  db.transaction(function(tx) {
    tx.executeSql('DROP TABLE IF EXISTS test_table');
    tx.executeSql('CREATE TABLE IF NOT EXISTS test_table (id integer primary key, data text, data_num integer)');
 
    // demonstrate PRAGMA: 
    db.executeSql("pragma table_info (test_table);", [], function(res) {
      console.log("PRAGMA res: " + JSON.stringify(res));
    });
 
    tx.executeSql("INSERT INTO test_table (data, data_num) VALUES (?,?)", ["test", 100], function(tx, res) {
      console.log("insertId: " + res.insertId + " -- probably 1");
      console.log("rowsAffected: " + res.rowsAffected + " -- should be 1");
 
      db.transaction(function(tx) {
        tx.executeSql("select count(id) as cnt from test_table;", [], function(tx, res) {
          console.log("res.rows.length: " + res.rows.length + " -- should be 1");
          console.log("res.rows.item(0).cnt: " + res.rows.item(0).cnt + " -- should be 1");
        });
      });
 
    }, function(e) {
      console.log("ERROR: " + e.message);
    });
  });
}

**Comment by brody2consult** [comment]
@Rombersoft I think your issue is different from the issue under discussion. It would be best to raise a new question in https://github.com/litehelpers/Cordova-sqlite-help/issues in case of this kind of issue. I also find your code hard to follow. Please enclose with
```
    ```js
    ```
```
to make it easier to follow.

One thing evident is that PRAGMA statements should never be executed within transactions. This is the fault of the sample in <https://github.com/litehelpers/Cordova-sqlite-storage#sample-with-pragma-feature>. I just raised https://github.com/litehelpers/Cordova-sqlite-help/issues/24 to fix this sample and document this as a pitfall.
